openwisp_utils/bots/ci_failure/bot.py

```
@@ -0,0 +1,483 @@
+                return
+            else:
+                # We have PR context, proceed with analysis
+                build_logs = self.get_build_logs()
+                pr_diff = self.get_pr_diff()
+                if not build_logs and not pr_diff:
+                    print("No build logs or PR diff found, using fallback response")
+                    message = self.fallback_response()
+                else:
+                    print("Analyzing failure with Gemini AI...")
+                    message = self.analyze_with_gemini(build_logs, pr_diff)
+        except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)